src/gh_requests/pull_requests.py

```diff
@@ -0,0 +1,117 @@
+        filter_params += f" review-requested:{pull_request_query.review_requested_user}"
+    if pull_request_query.reviewed_by:
+        filter_params += f" reviewed-by:{pull_request_query.reviewed_by}"
+    if pull_request_query.author:
+        filter_params += f" author:{pull_request_query.author}"
+    if pull_request_query.title:
+        filter_params += f" in:title {pull_request_query.title}"
+
+    issues = list(gh.search_issues(query=filter_params))
+    return list(_fetch_prs_concurrently(issues, pull_request_query.check_github_actions))
+
+
+def fetch_updated_pull_requests(
+    current_pull_requests: list[PullRequest], modified_pull_requests: list[PullRequest]
+) -> list[PullRequest]:
+    gh = Github(st.session_state.token)
+    updated_prs = _fetch_updated_prs_concurrently(gh, modified_pull_requests)
+    return list(_combine_prs_with_precedence(current_pull_requests, updated_prs))
+
+
+def _fetch_prs_concurrently(
+    issues: list[Issue],
+    check_github_actions: bool,
+) -> Iterator[PullRequestWithDetails]:
+    with ThreadPoolExecutor(max_workers=4) as task_pool:
+        return task_pool.map(partial(_fetch_pr, check_github_actions), issues)
+
+
+def _fetch_pr(check_github_actions: bool, issue: Issue) -> PullRequestWithDetails:
+    return _add_pr_details(issue.as_pull_request(), check_github_actions)
+
+
+def _fetch_updated_prs_concurrently(
+    gh: Github,
+    prs: list[PullRequest],
+    check_github_actions: bool = False,
+) -> Iterator[PullRequestWithDetails]:
+    with ThreadPoolExecutor(max_workers=4) as task_pool:
+        return task_pool.map(partial(_fetch_updated_pr, gh, check_github_actions), prs)
+
+
+def _fetch_updated_pr(gh: Github, check_github_actions: bool, pr: PullRequest) -> PullRequest:
+    return _add_pr_details(gh.get_repo(pr.base.repo.full_name).get_pull(pr.number), check_github_actions)
+
+
+def _add_pr_details(pr: PullRequest, check_github_actions: bool) -> PullRequestWithDetails:
+    return PullRequestWithDetails(
+        pr=pr,
+        title=pr.title,
+        user=pr.user.login,
+        name=pr.base.repo.full_name,
+        number=pr.number,
+        html_url=pr.html_url,
+        needs_rebase=not pr.mergeable,
+        is_approved=_is_approved(pr),
+        github_action_checked=check_github_actions,
+        is_ready_to_merge=_is_ready_to_merge(pr) if check_github_actions else False,
+        is_merged=pr.merged,
+    )
+
+
+def _combine_prs_with_precedence(a: Iterable[PullRequest], b: Iterable[PullRequest]) -> Iterable[PullRequest]:
+    # PRs in B take precedence over ones in A, as they follow in the chain
+    pr_map = {pr.number: pr for pr in chain(a, b)}
+    return pr_map.values()
+
+
+def _is_approved(pr: PullRequest) -> bool:
+    approved_reviews = 0
+
+    for review in pr.get_reviews():
+        if review.state == "APPROVED":
+            approved_reviews += 1
+        elif review.state == "CHANGES_REQUESTED":
+            return False
+    return approved_reviews > 0
```

Comment:
**praise:** good thinking with discarding any approvals if another review requested changes 🌟

Reply:
Oh I was just using this to fill in the form view but now that you mentioned it I addded some logic to avoid attempting to merge prs that aren't suppose to be.